### PR TITLE
*: update client-go

### LINKED ISSUE #33537: Make TiDB be compatible with latest client-go
## Enhancement

and resolve https://github.com/tikv/client-go/issues/455

### PR DESCRIPTION
**Pull request by youjiali1995**
close pingcap/tidb#33537

Signed-off-by: youjiali1995 <zlwgx1023@gmail.com>

### What problem does this PR solve?

Issue Number: close #33537

Problem Summary:

Make TiDB be compatible with latest client-go.

### What is changed and how it works?

Update client-go.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -751,6 +751,8 @@ github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2 h1:mbAskLJ0oJf
 github.com/tiancaiamao/appdash v0.0.0-20181126055449-889f96f722a2/go.mod h1:2PfKggNGDuadAa0LElHrByyrz4JPZ9fFx6Gs7nx7ZZU=
 github.com/tikv/client-go/v2 v2.0.1-0.20220321123529-f4eae62b7ed5 h1:XcvL365oWoFAdprI2H23vvFQYHc2MBQ/CcWHvkvIzGk=
 github.com/tikv/client-go/v2 v2.0.1-0.20220321123529-f4eae62b7ed5/go.mod h1:0scaG+seu7L56apm+Gjz9vckyO7ABIzM6T7n00mrIXs=
+github.com/tikv/client-go/v2 v2.0.1-0.20220328083738-8489c3e8c3d9 h1:TgSywPECh84B5S6Z97jhMdQVrezPwUbOKbCZFie2kPI=
```
Comment (jackysp) [inline go.sum:754]: missing `go mod tidy`?

Comment (youjiali1995) [reply go.sum:754]: Oh, yes.

**Review by jackysp — APPROVED** [review]
LGTM

**Review by sticnarf — APPROVED** [review]
LGTM

**Pull request merged**